Compilation error: Cannot open include file: 'dlfcn.h'

**Describe the bug**
When building for windows on windows with VS19 I get the following compilation error:

```
20:58:05  C:\Users\****local\workspace\422-use-oqs-provider-main-branch\submodules\linphone-sdk\external\liboqs\src\common\ossl_helpers.c(7): fatal error C1083: Cannot open include file: 'dlfcn.h': No such file or directory
```

**Environment (please complete the following information):**
 - OS: windows, VS 19
 - OpenSSL version: -
 - liboqs version: main

--- a/src/common/ossl_helpers.c
+++ b/src/common/ossl_helpers.c
@@ -4,7 +4,9 @@
 #define OQS_OSSL_NO_EXTERN 1
 #include "ossl_helpers.h"
 #include <assert.h>
+#if !defined(_WIN32)
 #include <dlfcn.h>
+#endif
 
 #if OPENSSL_VERSION_NUMBER >= 0x30000000L
 #if defined(OQS_USE_PTHREADS)

